test/framework/src/main/java/org/elasticsearch/test/gateway/TestGatewayAllocator.java

```diff
@@ -126,6 +131,19 @@ public void allocateUnassigned(
         innerAllocatedUnassigned(allocation, primaryShardAllocator, replicaShardAllocator, shardRouting, unassignedAllocationHandler);
     }
 
+    @Override
+    public AllocateUnassignedDecision explainUnassignedShardAllocation(ShardRouting unassignedShard, RoutingAllocation routingAllocation) {
```

Comment:
I needed to move this method over from the base class, because the subclass uses different allocators. The Gateway allocator keeps track of the unassigned shards, so it's the one that would explain the shard movement.